lava-v2-jobs-from-api.py

```diff
@@ -157,7 +157,8 @@ def get_job_params(config, test_config, defconfig, opts, build, plan):
         'arch_defconfig': opts['arch_defconfig'],
         'fastboot': str(device_type.get_flag('fastboot')).lower(),
         'priority': config.get('priority'),
-        'device_type': device_type.type_name,
+        'device_type': device_type.name,
+        'device_type_name': device_type.type_name,
```

Comment:
One problem with `type` is that it's a built-in class name in Python.  Maybe it's safe as an object attribute to have it, but it's not ideal as a variable.

The issue comes from the fact that there isn't always a one-to-one mapping between device type entries in the YAML config and actual device types.  QEMU is the best example of this, and the only real case at the moment in the YAML config.  Other potential cases would be for FPGA platforms where you can choose the type of CPU to load at boot time, or any kind of emulator.  For all the other current silicon-based devices, we can just use their "real" names in YAML which gets directly used in LAVA jobs.

So we need to be able to define this in practice, and the way it's done now is by using the `type` attribute in YAML to define the actual platform type as understood by LAVA labs (so "qemu" for all QEMU variants).  I agree that it would be better to use the same name in the Python attribute as in the YAML config, but to be fair I'm not entirely convinced by `type` either as that's very vague.  Maybe we should call it `lab_type` instead, or `base_type`.  It will also need to be documented once the YAML schema has settled down, which should help clarifying any doubt around the meaning of these attributes.